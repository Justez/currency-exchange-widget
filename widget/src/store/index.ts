import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import createRootReducer from './root-reducer';
import sagas from './root-saga';

const configureStore = () => {
  const sagaMiddleware = createSagaMiddleware();

  const middlewares = [
    sagaMiddleware,
  ];

  const instance = createStore(
    createRootReducer(),
    composeWithDevTools(applyMiddleware(...middlewares)),
  );

  sagaMiddleware.run(sagas);

  return instance;
};

export type State = ReturnType<typeof store.getState>;

const store = configureStore();

export default store;
