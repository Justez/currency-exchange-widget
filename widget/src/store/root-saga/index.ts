export { default } from './root-saga';