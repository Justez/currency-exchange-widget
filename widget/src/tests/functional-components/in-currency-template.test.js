import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from 'store';
import InCurrencyTemplate from 'functional-components/in-currency-template';

const component = (
    <Provider store={store}>
        <InCurrencyTemplate />
    </Provider>
)

describe('In Currency template snapshots', () => {
    it('renders in-currency-template', () => {
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});