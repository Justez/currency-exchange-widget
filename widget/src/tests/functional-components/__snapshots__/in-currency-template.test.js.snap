// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`In Currency template snapshots renders in-currency-template 1`] = `
<div
  className="MuiBox-root MuiBox-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 MuiGrid-direction-xs-column MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-between"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="choose-in-currency"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="choose-in-currency"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <h5
                  className="MuiTypography-root MuiTypography-h5"
                >
                  GBP
                </h5>
              </div>
              <input
                onAnimationStart={[Function]}
                type="hidden"
                value="gbp"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              Balance
               
              102.22
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                style={
                  Object {
                    "height": "11",
                    "marginLeft": "4px",
                    "width": "11",
                  }
                }
                viewBox="0 0 8 8"
              >
                <path
                  d="M3 0c-.62 0-1.16.26-1.5.69-.34.43-.5.99-.5 1.56 0 .69.16 1.25.25 1.75h-1.25v1h1.22c-.11.45-.37.96-1.06 1.66l-.16.13v1.22h6v-1h-4.91c.64-.73.98-1.4 1.13-2h1.78v-1h-1.72c-.08-.68-.28-1.24-.28-1.75 0-.39.11-.73.28-.94.17-.21.37-.31.72-.31.39 0 .61.11.75.25s.25.36.25.75h1c0-.58-.17-1.1-.53-1.47-.37-.37-.89-.53-1.47-.53z"
                />
              </svg>
            </p>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-3 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "height": "18px",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <input
            className="MuiInputBase-input MuiInput-input makeStyles-input-4 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
            disabled={false}
            id="amount-received"
            max={102.22}
            maxLength={7}
            min={0}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="number"
            value={12}
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              style={
                Object {
                  "height": "11",
                  "marginLeft": "4px",
                  "width": "11",
                }
              }
              viewBox="0 0 8 8"
            >
              <path
                d="M3 0c-.62 0-1.16.26-1.5.69-.34.43-.5.99-.5 1.56 0 .69.16 1.25.25 1.75h-1.25v1h1.22c-.11.45-.37.96-1.06 1.66l-.16.13v1.22h6v-1h-4.91c.64-.73.98-1.4 1.13-2h1.78v-1h-1.72c-.08-.68-.28-1.24-.28-1.75 0-.39.11-.73.28-.94.17-.21.37-.31.72-.31.39 0 .61.11.75.25s.25.36.25.75h1c0-.58-.17-1.1-.53-1.47-.37-.37-.89-.53-1.47-.53z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary Mui-disabled MuiButton-fullWidth Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Exchange
        </span>
      </button>
    </div>
  </div>
</div>
`;
